uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -77,23 +77,24 @@ define([
        * @return {Promise}
        */
       function saveAndDeleteComments () {
+        var leaveRequestId = this.id;
         var promises = [];
-        var self = this;
-
-        // Save comments which dont have an ID
-        self.comments.map(function (comment, index) {
-          if (!comment.comment_id) {
-            // IIFE is created to keep actual value of 'index' when promise is resolved
-            (function (index) {
-              promises.push(LeaveRequestAPI.saveComment(self.id, comment)
-                .then(function (commentData) {
-                  self.comments[index] = commentData;
-                }));
-            })(index);
-          } else if (comment.toBeDeleted) {
-            promises.push(LeaveRequestAPI.deleteComment(comment.comment_id));
-          }
-        });
+
+        // Pushing a chain of API calls to create new comments sequentially
+        promises.push($q.sequence(this.comments.filter(function (comment) {
+          return !comment.comment_id;
+        }).map(function (comment) {
+          return function () {
+            return LeaveRequestAPI.saveComment(leaveRequestId, comment);
```

Comment:
previously the comments array would be updated with the new comment information (which would have the *comment_id* property). This probably was done in case the modal would stay open after saving it and the user would make changes and save again. In that scenario the comments would not be saved twice. It seems now they do, am I right?